core-contracts/AssetManager/src/intTest/java/network/balanced/score/core/asset/manager/AssetManagerIntTest.java

```diff
@@ -0,0 +1,265 @@
+package network.balanced.score.core.asset.manager;
+
+import com.eclipsesource.json.JsonArray;
+import foundation.icon.score.client.DefaultScoreClient;
+import foundation.icon.score.client.RevertedException;
+import foundation.icon.xcall.NetworkAddress;
+import network.balanced.score.lib.interfaces.AssetManagerMessages;
+import network.balanced.score.lib.interfaces.tokens.SpokeTokenScoreClient;
+import network.balanced.score.lib.test.integration.Balanced;
+import network.balanced.score.lib.test.integration.BalancedClient;
+import network.balanced.score.lib.test.integration.Env;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import score.Address;
+
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Map;
+
+import static network.balanced.score.lib.test.integration.BalancedUtils.createParameter;
+import static network.balanced.score.lib.test.integration.BalancedUtils.createSingleTransaction;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class AssetManagerIntTest {
+
+    private static final Env.Chain chain = Env.getDefaultChain();
+    private static Balanced balanced;
+    private static BalancedClient owner;
+    private static BalancedClient reader;
+    private static BalancedClient user;
+
+    public String INJ_NID = "0x4.INJ";
+    String injAsset1Address = "inj1x3";
+
+    NetworkAddress injSpoke = new NetworkAddress(INJ_NID, "inj1x1");
+
+    @BeforeAll
+    static void setup() throws Exception {
+        balanced = new Balanced();
+        balanced.setupBalanced();
+
+        owner = balanced.ownerClient;
+        reader = balanced.newClient(BigInteger.ZERO);
+        user = balanced.newClient();
+    }
+
+    @Test
+    @Order(1)
+    void linkToken() {
+        // Arrange
+        NetworkAddress tokenNetworkAddress = new NetworkAddress(INJ_NID, injAsset1Address);
+        Address linkTokenAddress = owner.sicx._address();
+
+
+        //Act
+        RevertedException thrown = assertThrows(
+                RevertedException.class, () ->  owner.governance.execute(createSingleTransaction(balanced.assetManager._address(), "linkToken",
+                    new JsonArray().add(createParameter(tokenNetworkAddress.toString())).add(createParameter(linkTokenAddress))).toString())
+        );
+        assertTrue(thrown.getMessage().contains("UnknownFailure"));
```

Comment:
This revert message check can be removed. Might be better to name executable to explain error
```suggestion
        //Act
        JsonArray params =  new JsonArray()
             .add(createParameter(tokenNetworkAddress.toString()))
             .add(createParameter(linkTokenAddress))).toString());
        Executable linkTokenWithoutAssetManager () -> owner.governance.execute(createSingleTransaction(balanced.assetManager._address(), "linkToken", params);
         assertThrows(RevertedException.class,   linkTokenWithoutAssetManager);
```

Reply:
Same kind of structure can be applied in more places, Try to split the function calls so the become a bit clearer.
No need to assert messages if there are no info to be gained from them

Reply:
In general try to create well named variables where possible to make code easier to read